### PR TITLE
adding test for bafrio, fix some code coverage issues

### LINKED ISSUE #56: byteswap.c has system pre-processor define code, but always sets LINUX anyway...
In byteswap.c we have code like this:

```
#ifdef LINUX
  void byteswap_
         (char *data, int *nbyte, int *nnum) {
#endif
#ifdef IBM4
  void byteswap
         (char *data, int *nbyte, int *nnum) {
#endif
#ifdef IBM8
  void byteswap
         (char *data, long long int *nbyte, long long int *nnum) {
#endif

```

But we also have this at the top of the file:

`#define LINUX /**< Define LINUX. */
`

So LINUX is the only one we need...

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #42 
Part of #33
Part of #38
Fixes #56

Adding a test for barfio.f90.

Also improving doxygen.

Also fixing code coverage by adding CFLAGS.

Also removed some unused pre-processor ifdefs from byteswap.c.

### CODE REVIEW

File: src/bafrio.f90 [hunk 1]
```diff
@@ -138,7 +138,7 @@ SUBROUTINE BAFRINDEXL(LU,IB,LX,IX)
 ! - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
       END SUBROUTINE BAFRINDEXL
 
-!> This subprogram is calling bafread to reas an unformatted fortran
+!> This subprogram calls bafread() to reas an unformatted fortran
```
Comment (BrianCurtis-NOAA) [inline src/bafrio.f90]: to read* an

File: src/bafrio.f90 [hunk 2]
```diff
@@ -138,7 +138,7 @@ SUBROUTINE BAFRINDEXL(LU,IB,LX,IX)
...
 !> record. The difference between bafrread() and bafrreadl() is the
 !> kind type of intergers in the argument list.
```
Comment (BrianCurtis-NOAA) [inline src/bafrio.f90]: integers*

File: src/bafrio.f90 [hunk 3]
```diff
@@ -230,21 +230,21 @@ SUBROUTINE BAFRREADL(LU,IB,NB,KA,A)
 ! - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
       END SUBROUTINE BAFRREADL
 
-!> This subprogram is calling bafrwrite to write an unformatted
-!> fortran record. The difference between bafrwrite and bafrwritel is
-!> the kind type of intergers in the argument list
```
Comment (BrianCurtis-NOAA) [inline src/bafrio.f90:235]: integers*

File: src/bafrio.f90 (hunk 2 above)
Comment (edwardhartnett) [reply src/bafrio.f90]: Awkward phrasing but not something I'm going to worry about. I think in time we'll be able to eliminate this code duplication.

File: src/bafrio.f90 (hunk 3 above)
Comment (edwardhartnett) [reply src/bafrio.f90:235]: Oh it's misspelled. I get it. OK, fixed.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Pull request merged**